eng/scripts/MavenPackaging.ps1

```diff
@@ -153,3 +153,66 @@ function Get-FilteredMavenPackageDetails([string]$ArtifactDirectory, [string]$Gr
 
   return $filteredPackageDetails
 }
+
+# Compare the contents of SHA hash files for a local repository to the same
+# files in Maven Central or Azure DevOps. Returns $false if none of the hash
+# files exist in the remote, $true if all hashes exist and match the local
+# hash, otherwise throws.
+function Test-ReleasedPackage([string]$RepositoryUrl, [MavenPackageDetail]$PackageDetail, [string]$BearerToken) {
+  if ($RepositoryUrl -match "^https://pkgs.dev.azure.com/azure-sdk/\b(internal|public)\b/*") {
+    if(!$BearerToken) {
+      throw "BearerToken required for Azure DevOps package feeds"
+    }
+    
+    $baseUrl = "https://pkgs.dev.azure.com/azure-sdk/internal/_packaging/azure-sdk-for-java-pr/maven/v1"
+    $algorithm = "sha256"
+    $headers = @{ Authorization="BEARER $BearerToken" }
+  }
+  elseif ($RepositoryUrl -match "^https://oss.sonatype.org/service/local/staging/deploy/maven2") {
+    $baseUrl = "https://repo1.maven.org/maven2"
+    $algorithm = "sha1"
+  }
+  else {
+    throw "Repository URL must be either an Azure Artifacts feed, or a SonaType Nexus feed."
+  }
+
+  $packageUrl = "$baseUrl/$($PackageDetail.GroupId.Replace('.', '/'))/$($PackageDetail.ArtifactID)/$($PackageDetail.Version)"
+  
+  # Count the number of remote hashes found
+  $remoteCount = 0
+
+  # Count the number of remote hashes that match their local hash
+  $matchCount = 0
+
+  foreach($artifact in $PackageDetail.AssociatedArtifacts) {
```

Comment:
Do you think it is worth checking all of them? If it is cheap enough I guess it doesn't matter but it might not really be needed.

Reply:
It's really cheap. ~1 second to check 6 files.

Reply:
That is when things are working :) The more times we download something the more likely we end up in a networking issue we now have to retry.

Reply:
I'd like to check the availability of the different Sonotype apis. repo1.maven.org is currently CDNed at Fastly.   I would expect this to not suffer from the availability issues at oss.sonatype.org.